### DISCUSSION TITLE
Using the same algorithm for the initial key exchange and the additional key exchange

### DISCUSSION THREAD

**Discussion by yamus-didine** [strongswan/strongswan#1819] [Q&A] [answered]
Hello.
I am using strongswan v 6.0.0beta3. I am trying different scenarios for the IKE proposals.
My inquiry is that I want to use an algorithm for the additional key exchange similar to the algorithm used for the initial key exchange. For example, I want the IKE proposal to be like this
`proposals = aes128gcm128-sha256-x25519-ke1_x25519` 

However, I got this log (on the initiator side)
![same_kem_alice](https://github.com/strongswan/strongswan/assets/134303261/50d99d56-9818-4f0d-96e1-7d05183c3b13)

And this log (on the responder side)
![same_kem_bob](https://github.com/strongswan/strongswan/assets/134303261/4e9f925a-c136-49e1-80b8-55f18494c226)

Is there a way to overcome this issue?

**Comment by tobiasbrunner** [comment, marked as answer]
> Is there a way to overcome this issue?

No, RFC 9370, section 2.2.1 states:
```
However, for the ADDKE Transform Types, the responder's choice MUST NOT contain
duplicated algorithms (those with an identical Transform ID and
attributes), except for the Transform ID of NONE.
...
If the responder is unable to select algorithms that are not duplicated
for each proposed key exchange (either because the proposal contains
too few choices or due to the local policy restrictions on using the
proposed algorithms), then the responder MUST reject the message with
an error notification of type NO_PROPOSAL_CHOSEN.
```
